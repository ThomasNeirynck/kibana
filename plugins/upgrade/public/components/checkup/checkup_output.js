import React from 'react';

import { IndexDeprecations } from './index_deprecations';
import { ClusterDeprecations } from './cluster_deprecations';
import { NodeDeprecations } from './node_deprecations';


export function CheckupOutput({ className, output }) {
  return (
    <div className={ className }>
      <IndexDeprecations
        className="kuiVerticalRhythm"
        deprecations={ output.index_settings }
      />
      <NodeDeprecations
        className="kuiVerticalRhythm"
        deprecations={ output.node_settings }
      />
      <ClusterDeprecations
        className="kuiVerticalRhythm"
        deprecations={ output.cluster_settings }
      />
    </div>
  );
}

CheckupOutput.propTypes = {
  className: React.PropTypes.string,
  output: React.PropTypes.shape({
    cluster_settings: React.PropTypes.array,
    index_settings: React.PropTypes.object,
    node_settings: React.PropTypes.array,
  }),
};

CheckupOutput.defaultProps = {
  className: null,
  output: {},
};
