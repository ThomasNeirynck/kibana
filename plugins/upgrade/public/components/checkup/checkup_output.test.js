import { shallow } from 'enzyme';
import React from 'react';

import { CheckupOutput } from './checkup_output';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('CheckupOutput', () => {
  test('renders the className property', () => {
    const component = (
      <CheckupOutput className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders Index-, Node- and ClusterDeprecations with the given deprecations', () => {
    const output = {
      index_settings: {
        index1: [],
      },
      node_settings: [
        {
          level: 'warning',
        },
      ],
      cluster_settings: [
        {
          level: 'critical',
        },
      ],
    };
    const component = (
      <CheckupOutput output={output}/>
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
