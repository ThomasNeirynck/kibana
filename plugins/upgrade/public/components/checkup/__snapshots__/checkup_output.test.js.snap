// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckupOutput renders Index-, Node- and ClusterDeprecations with the given deprecations 1`] = `
<div
  className={null}
>
  <IndexDeprecations
    className="kuiVerticalRhythm"
    deprecations={
      Object {
        "index1": Array [],
      }
    }
  />
  <NodeDeprecations
    className="kuiVerticalRhythm"
    deprecations={
      Array [
        Object {
          "level": "warning",
        },
      ]
    }
  />
  <ClusterDeprecations
    className="kuiVerticalRhythm"
    deprecations={
      Array [
        Object {
          "level": "critical",
        },
      ]
    }
  />
</div>
`;

exports[`CheckupOutput renders the className property 1`] = `
<div
  className="myClass"
>
  <IndexDeprecations
    className="kuiVerticalRhythm"
    deprecations={Object {}}
  />
  <NodeDeprecations
    className="kuiVerticalRhythm"
    deprecations={Array []}
  />
  <ClusterDeprecations
    className="kuiVerticalRhythm"
    deprecations={Array []}
  />
</div>
`;
